### ISSUE TITLE
Improve CheckInServices configuration changes

### ISSUE THREAD

**Issue by klinker24** [net-angel/netangel-protection-android#19]
When the `CheckInService` runs, it does a check to see if any configuration (vpn enabled, battery saver, pause vpn) have changed from the web, then it broadcasts those changes to anyone listening.

Unfortunately, the only think listening, right now, is the `MainActivity`, which will get destroyed and probably won't hear any of these changes unless the user is currently looking at it.

This configuration change logic should be moved to the `VpnStatusService` instead, where it will always be listening.

Within that service, right now, all we should do is check to see whether the VPN should still be enabled or not. In the future, it should also manage the battery saver feature.

**Comment by eterry1388** [comment]
How it is set up today, the app checks the server to see if there are any changes.  This is not the best way to do it as a change on the dashboard would take some time before it is applied on the device.  Is it better and is there a way to have the dashboard push the setting change to the app rather than have the app pull from the server?

**Comment by klinker24** [comment]
Yes, Google has a service called [Firebase](https://firebase.google.com/docs/cloud-messaging/) that allows for push notifications from the web to a device. I have used it in the past, it also works for iOS.

Pretty much, we generate a unique device id on the phone, and save that to their account. When something changes on the web, we can use the backend to push those changes to the device immediately, instead of polling the backend from the phone.

I have used this several times in the past, it is non-trivial to set up, unfortunately, there would be a big effort to get this into the app, but it probably would be worth it in the long run. Firebase is a free service
